documentation/modules/oauth/con-oauth-authorization-keycloak-example.adoc

```diff
@@ -0,0 +1,481 @@
+  oauth.client.id="kafka-cli" \
+  oauth.token.endpoint.uri="http://keycloak:8080/auth/realms/kafka-authz/protocol/openid-connect/token" ;
+sasl.login.callback.handler.class=io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler
+EOF
+```
+
+Note that we use the `kafka-cli` public client for the `oauth.client.id` in the `sasl.jaas.config`.
+Since that is a public client it does not require any secret.
+We can use this because we authenticate with a token directly (in this case a refresh token is used to request an access token behind the scenes which is then sent to Kafka broker for authentication, and we already did the authentication for the refresh token).
+
+
+Let's now try to create the `x_messages` topic:
+
+    bin/kafka-topics.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --command-config /tmp/bob.properties \
+      --topic x_messages --create --replication-factor 1 --partitions 1
+
+The operation should succeed (you can ignore the warning about periods and underscores).
+
+We can list the topics:
+
+    bin/kafka-topics.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --command-config /tmp/bob.properties --list
+
+If we try the same as `team-a-client` or `team-b-client` we will get different responses.
+
+    bin/kafka-topics.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --command-config /tmp/team-a-client.properties --list
+    bin/kafka-topics.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --command-config /tmp/team-b-client.properties --list
+
+Roles `Dev Team A`, and `Dev Team B` both have `Describe` permission on topics that start with 'x_', but they cannot see the other team's topics as they do not have `Describe` permissions on them.
+
+We can now again try to produce to the topic as `team-a-client`.
+
+```
+bin/kafka-console-producer.sh --broker-list my-cluster-kafka-bootstrap:9092 --topic x_messages \
+  --producer.config /tmp/team-a-client.properties
+Message 1
+Message 2
+Message 3
+```
+
+This works.
+
+If we try the same as `team-b-client` it should fail.
+
+```
+bin/kafka-console-producer.sh --broker-list my-cluster-kafka-bootstrap:9092 --topic x_messages \
+  --producer.config /tmp/team-b-client.properties
+Message 4
+Message 5
+```
+
+We get an error - `Not authorized to access topics: [x_messages]`.
+
+But `team-b-client` should be able to consume messages from the `x_messages` topic:
+
+    bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic x_messages \
+      --from-beginning --consumer.config /tmp/team-b-client.properties --group x_consumer_group_b
+
+Whereas `team-a-client` does not have permission to read, even though they can write:
+
+    bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic x_messages \
+      --from-beginning --consumer.config /tmp/team-a-client.properties --group x_consumer_group_a
+
+We get a `Not authorized to access group: x_consumer_group_a` error.
```

Comment:
```suggestion
Conversely, `team-a-client` does not have permission to read from topic `x_messages`, even though it can write, so the following operation returns an error:

[source,shell]
----
bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic x_messages \
  --from-beginning --consumer.config /tmp/team-a-client.properties --group x_consumer_group_a
----
```